applications/P2PStateChannels.md

```diff
@@ -0,0 +1,254 @@
+
+4) DisputeHandler - 
+Whenever a peer detects deviation from the protocl, they initiate a dispute by going through the DisputeHandler, which creates a dispute on-chain asking the DLT to enforce a correct state. To initiate a dispute a peer provides the last known finalized transaction (state) + all the transactions till the latest one the peer signed (confirmed) (this is at most N transactions in total - see Virtual Voting). After initiating a dispute, the peer stops signing transactions for the given fork. Due to the chain not having all the data and ideally not having any, the disputes are not final and have a challenge period where they can be contested. If successfully challenged (by providing more data), the disputter is fully slashed, a correct state is enforced and the challenge window prolonged. All honest peers check the disputed data and without delay challenge it if possible, otherwise they continue building on the fork and let the challenge period expire. More details on Dispute Logic down below when we cover on-chain logic.
+ 
+5) DLTEventListener -
+Observes the DLT for relative events and passes them to the StateManager.
+In V1 the blockchain interaction will be done through a provider, which the peers trust. In V2 we’ll explore options of making peers ultra light clients so they can access the DLT directly and increase resilience (the bootstrapping problem makes this slight optimization not worth considering in V1, since some node has to always be trusted to provide the data).
+
+6) AStateMachine - 
+Abstract state machine that defines the base class/contract of compatible state machines. While any state and state transition can be implemented and thus supporting arbitrary state machines, the system does require a minimal interface with some minimal properties.
+
+7) StateObserver -
+Observes the concrete state machine and allows for other systems to subscribe for updates when the state changes. Analogous to on-chain events.
+
+The main responsibilities of on-chain smart contracts are: 
+1) Agreement verification - Verifies that all participants agree on a given state,
+2) Dispute resolution (shared security) - Enforcing a correct state of the state machine if peer-to-peer agreements fail,
+3) Settlement - final settlement of any assets that entered into the state channel,
+4) Data Availability - Guaranteeing data availability for both state and transaction calldata if peer-to-peer agreements fail,
+5) Liveness guarantees - In the case that the state machine can’t progress peer-to-peer, the DLT can progress it and remove any participants who failed to progress it within timeout,
+6) Objective clock - provides a single global clock (timestamp) that’s uncontested and accepted as correct/true unconditionally. Peers sync their local logical clock to the DLT clock to mimic the DLT perception of time,
```

Comment:
On 1+2+6.  Across all applications, the off-chain logic progresses by unanimous consensus, but individual steps have an author, and the others merely need to acknowledge them, while on-chain these acknowledgement steps can be replaced by a clock.  There is frequently an application specific part here too, in that if some turn logic dictates the next action's author, then if they abandon the channel then there must be some default action, even if only to abort for everyone.

Reply:
yes